cyclonedx/output/serializer/json.py

```diff
@@ -89,3 +93,10 @@ def default(self, o: Any) -> Any:
 
         # Fallback to default
         super().default(o=o)
+
+    @classmethod
+    def serialize_component_dependencies(cls, c: Component) -> Dict[str, Union[str, List[str]]]:
```

Comment:
The issue here is that `bom-ref`'s for Components are not guaranteed to the PURLs.

Each Component now has a `bom-ref` attribute which should be used.